syntax = "proto3";

service File {
  rpc download(FileDownloadReq) returns (stream FileDownloadRsp) {}
  rpc list(ListReq) returns (stream ListRsp) {}
  rpc upload(stream FileUploadReq) returns (UploadRsp) {}
}

message FileDownloadRsp {
  bytes buffer = 1;
}

message FileDownloadReq {
  string name = 1;
}

message ListReq {
}

message ListRsp {
  string name = 1;
  int64 size = 2;
}
message FileUploadReq {
    bytes buffer = 1;
    string name = 2;
}

message UploadRsp {
    string message = 1;
}